sdk/ai/azure-ai-agents/azure/ai/agents/models/_patch.py

```diff
@@ -1689,15 +1842,80 @@ def get_definitions_and_resources(self) -> Dict[str, Any]:
             "tools": self.definitions,
         }
 
+    @overload
+    def get_tool(self, tool_type: Type[McpTool]) -> McpTool:
+        """
+        Get an MCP tool from the tool set.
+
+        :param tool_type: The MCP tool type to get.
+        :type tool_type: Type[McpTool]
+        :return: The MCP tool.
+        :rtype: McpTool
+        """
+        ...
+
+    @overload
+    def get_tool(self, tool_type: Type[McpTool], *, server_label: str) -> McpTool:
+        """
+        Get an MCP tool with a specific server label from the tool set.
+
+        :param tool_type: The MCP tool type to get.
+        :type tool_type: Type[McpTool]
+        :keyword server_label: The server label of the specific MCP tool to get.
+        :paramtype server_label: str
+        :return: The MCP tool with the specified server label.
+        :rtype: McpTool
+        """
+        ...
+
+    @overload
     def get_tool(self, tool_type: Type[ToolT]) -> ToolT:
         """
         Get a tool of the specified type from the tool set.
 
-        :param Type[Tool] tool_type: The type of tool to get.
+        :param tool_type: The type of tool to get.
+        :type tool_type: Type[Tool]
         :return: The tool of the specified type.
         :rtype: Tool
-        :raises ValueError: If a tool of the specified type is not found.
         """
+        ...
+
+    def get_tool(self, tool_type: Type[ToolT], **kwargs) -> ToolT:
```

Comment:
Let's avoid using `**kwargs` unless we do a pass-through of the arguments and/or allow for a truly arbitrary set of parameter names (e.g. the way `dict` does).